src/shared/sam/localLambdaRunner.ts

```diff
@@ -352,22 +525,35 @@ export async function attachDebugger({
     }
 }
 
-export async function waitForDebugPort(
-    debugPort: number,
-    timeoutDuration: Timeout,
-    channelLogger: ChannelLogger
+/**
+ * Waits for a port to be in use.
+ *
+ * @param port  Port number to wait for
+ * @param timeout  Time to wait
+ * @param channelLogger  Logger
+ * @param isDebugPort  Is this a debugger port or a `sam local start-api` HTTP port?
+ */
+export async function waitForPort(
+    port: number,
+    timeout: Timeout,
+    channelLogger: ChannelLogger,
+    isDebugPort: boolean = true
 ): Promise<void> {
-    const remainingTime = timeoutDuration.remainingTime
+    const time = timeout.remainingTime
     try {
         // this function always attempts once no matter the timeoutDuration
-        await tcpPortUsed.waitUntilUsed(debugPort, SAM_LOCAL_PORT_CHECK_RETRY_INTERVAL_MILLIS, remainingTime)
+        await tcpPortUsed.waitUntilUsed(port, SAM_LOCAL_PORT_CHECK_RETRY_INTERVAL_MILLIS, time)
     } catch (err) {
-        getLogger().warn(`Timed out after ${remainingTime} ms waiting for port ${debugPort} to open: %O`, err as Error)
-
-        channelLogger.warn(
-            'AWS.samcli.local.invoke.port.not.open',
-            "The debug port doesn't appear to be open. The debugger might not succeed when attaching to your SAM Application."
-        )
+        getLogger().warn(`Timeout after ${time} ms: port was not used: ${port}`)
+        if (isDebugPort) {
```

Comment:
Nit: you can consolidate this if statement into one string: 
`channelLogger.warn('AWS.apig.portUnavailable', 'Failed to use {0} port: {1}', isDebugPort ? 'debugger' : 'API', port.toString())`

(unsure if we'd need to localize `debugger` and `API` as well, although they may make a nice addition to the generics section)

Reply:
I considered it. But it seems over-engineered for now.